magic-gds-drc/action.yml

```diff
@@ -0,0 +1,53 @@
+# Copyright 2021 Efabless Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Skywater PDK Libraries - Run Magic DRC on All Cells'
+description: |
+  This runs DRC using Magic on all SCL GDS files.
+
+  This also ensures that GDS files can be read by Magic.
+
+  Requires having checked out the repository.
+inputs:
+  match_directories:
+    description: |
+      Match directories under <repository_root>/cells/ with this regular expression.
+      Default: . (matches all)
+    required: false
+    default: '.'
+  known_bad:
+    description: |
+      List of cell names that are known bad. known_bad cells do not contribute
+      to the failure of this task. Comma,delimited,list
+      Default: Empty string (ignores nothing).
+    required: false
+    default: ''
+  acceptable_errors:
+    description: |
+      A file containing a newline delimited list of errors that are considered acceptable.
+    required: false
+    default: /dev/null
+runs:
+  using: docker
+  image: 'Dockerfile'
```

Comment:
I do not recommend building the container image on each execution of the Action. Instead, I suggest pre-building the image, and letting the Action use it from the registry. See https://github.com/mithro/actions-includes/blob/main/action.yml#L29.

If you did so, this action file would be unnecesary, since users might use the built-in syntax. See, for instance, https://github.com/tmeissner/formal_hw_verification/blob/master/.github/workflows/Test.yml#L40-L45.

Reply:
I think prebuilding the image is a good idea and I will get on it, but the action file acts as some kind of documentation if nothing else…

Reply:
In this case, the action is wrapping a container, with the entrypoint set to a Python script. Therefore, the API of the Action, the container and the script is exactly the same. That is, you did already document the API in https://github.com/google/skywater-pdk-actions/pull/14/files/667124a90d1d10e3a645d780a3c743459458d48a#diff-8787ed2cd84b43b2b05fb775a4d271e6b397598527605ed8a8a5fb4d5fe35f64R102-R124. The content of `action.yml` is duplication. In this context, it is a feature that users can execute the script locally, or they can use a container. I suggest stressing that feature, instead of adding a layer of complexity.

The use case would be different if the Action had any feature which is not a direct map of the script. See, for instance, https://github.com/VUnit/vunit_action/blob/master/action.yml. There, the Action is used for allowing the user to customise the image. Nonetheless, that does not apply to the use case here.

Reply:
Eh, fair enough. Will do.